OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseTabsDashboardFragment.kt

```diff
@@ -0,0 +1,886 @@
+        // The app get the sku details instantly, so add some wait to perform
+        // animation at least one cycle.
+        binding.accessError.setPrimaryButtonText(
+            ResourceUtil.getFormattedString(
+                resources,
+                R.string.label_upgrade_course_button,
+                AppConstants.PRICE,
+                skuDetails.price
+            ).toString()
+        )
+        binding.accessError.setPrimaryButtonListener {
+            iapAnalytics.trackIAPEvent(Analytics.Events.IAP_UPGRADE_NOW_CLICKED, "", "")
+            iapViewModel.startPurchaseFlow(skuDetails.sku)
+        }
+    }
+
+    private fun handleIAPException(errorMessage: ErrorMessage) {
+        var retryListener: DialogInterface.OnClickListener? = null
+
+        if (HttpStatus.NOT_ACCEPTABLE == (errorMessage.throwable as InAppPurchasesException).httpErrorCode) {
+            retryListener = DialogInterface.OnClickListener { _: DialogInterface?, _: Int ->
+                // already purchased course.
+                iapViewModel.iapFlowData.isVerificationPending = false
+                iapViewModel.iapFlowData.flowType = IAPFlowType.SILENT
+                showFullscreenLoader(iapViewModel.iapFlowData)
+            }
+        } else if (errorMessage.canRetry()) {
+            retryListener = DialogInterface.OnClickListener { _: DialogInterface?, _: Int ->
+                if (errorMessage.requestType == ErrorMessage.PRICE_CODE) {
+                    iapViewModel.initializeProductPrice(courseData.courseSku)
+                }
+            }
+        }
+        iapDialog.handleIAPException(this, errorMessage, retryListener, null)
+    }
+
+    private fun showFullscreenLoader(iapFlowData: IAPFlowData) {
+        // To proceed with the same instance of dialog fragment in case of orientation change
+        val fullscreenLoader = fullscreenLoader ?: newInstance(iapFlowData)
+        fullscreenLoader.show(childFragmentManager, FullscreenLoaderDialogFragment.TAG)
+    }
+
+    private fun onCloseClick(): View.OnClickListener {
+        return View.OnClickListener { requireActivity().finish() }
+    }
+
+    private fun onFindCourseClick(): View.OnClickListener {
+        return View.OnClickListener {
+            environment.analyticsRegistry.trackUserFindsCourses()
+            EventBus.getDefault().post(MoveToDiscoveryTabEvent(Screen.DISCOVERY))
+            requireActivity().finish()
+        }
+    }
+
+    private fun initCourseDateObserver() {
+        courseDateViewModel.courseDates.observe(viewLifecycleOwner, EventObserver {
+            if (it.courseDateBlocks != null) {
+                it.organiseCourseDates()
+                val outdatedCalenderId =
+                    isCalendarOutOfDate(
+                        requireContext(),
+                        accountName,
+                        calendarTitle,
+                        it.courseDateBlocks
+                    )
+                if (outdatedCalenderId != -1L) {
```

Comment:
Please verify, do we still need to check the out dated calendar as now the banner will be visible on all the tabs.

Reply:
Yeap, I discussed it with Moiz and he advised me not to change an existing feature. Also coordinated it with Saeed.